### PR TITLE
Create `agoricd` for Cosmos chain

### PR DESCRIPTION
**Pull request by michaelfig**
refs: Private issue

## Description

The advantage is `agoricd` is more mnemonic and easier to type, and since it is a Golang binary, it will perform just as well as `ag-cosmos-helper` but will shell out to `ag-chain-cosmos` itself when `agoricd start` is run.

The new binary's features are compatible with both `ag-chain-cosmos` and `ag-cosmos-helper`, and is intended to be a performant replacement for both of them.  Its default `--home` is `$HOME/.agoricd`.

### Documentation Considerations

We use the same Go dependencies as `ag-chain-cosmos` and `ag-cosmos-helper`, but this packaging is intended to be a frontend that replaces both those commands.  The old commands remain for those who are stuck in the mud.

We will need to update the validator guides for this change.

### CODE REVIEW

File: golang/cosmos/app/app.go [hunk 1]
```diff
@@ -365,7 +365,7 @@ func NewAgoricApp(
 
 	// This function is tricky to get right, so we build it ourselves.
 	callToController := func(ctx sdk.Context, str string) (string, error) {
-		if vm.IsSimulation(ctx) {
+		if sendToController == nil || vm.IsSimulation(ctx) {
```
Comment (JimLarson) [inline golang/cosmos/app/app.go]: Add to comment below: "is being simulated ... or we're deliberately running without SwingSet".

**Review by JimLarson — APPROVED** [review]

File: golang/cosmos/app/app.go (hunk 1 above)
Comment (michaelfig) [reply golang/cosmos/app/app.go]: I removed this loosey-goosey `nil` check.  Now the mechanism is different.

Comment (erights) [reply golang/cosmos/app/app.go]: OMG a `==`. Don't our lint rules prohibit it?

Comment (michaelfig) [reply golang/cosmos/app/app.go]: > OMG a `==`. Don't our lint rules prohibit it?

Yes they do, but this is Golang, not JS.

Comment (erights) [reply golang/cosmos/app/app.go]: Doh!

File: golang/cosmos/Makefile [hunk 2]
```diff
@@ -40,10 +40,11 @@ ldflags_helper = $(ldflags) \
 BUILD_FLAGS := -tags "$(build_tags)" -gcflags '$(gcflags)' -ldflags '$(ldflags)'
 BUILD_FLAGS_HELPER :=  -buildmode=exe -tags "$(build_tags)" -gcflags '$(gcflags)' -ldflags '$(ldflags_helper)'
 
-all: compile-helper compile-daemon
-compile-go: compile-helper compile-libdaemon
-compile-node: node-compile-gyp
+all: compile-chain
```
Comment (JimLarson) [inline golang/cosmos/Makefile:43]: Good.  "All" should be all.

**Review by JimLarson — APPROVED** [review]

**Pull request merged**